src/app/forms/custom-sections/plates/plates.component.ts

```diff
@@ -112,4 +126,35 @@ export class PlatesComponent implements OnInit, OnDestroy, OnChanges {
     }
     return '';
   }
+
+  validateTechRecordPlates(): void {
+    this.globalErrorService.clearErrors();
+    const plateFieldsErrorMessage = 'All fields marked plate are mandatory to generate a plate.';
+    const plateValidationTable = this.techRecord.techRecord_vehicleType === 'trl' ? trlRequiredFields : hgvRequiredFields;
+
+    if (this.cannotGeneratePlate(plateValidationTable)) {
+      this.viewportScroller.scrollToPosition([0, 0]);
+      this.globalErrorService.addError({ error: plateFieldsErrorMessage });
+      return;
+    }
+    this.store.dispatch(canGeneratePlate());
+    this.router.navigate(['generate-plate'], { relativeTo: this.route });
+  }
+
+  private cannotGeneratePlate(plateValidationTable: string[]) {
```

Comment:
I think this method could be refined a bit more, also think maybe the method could be renamed

Reply:
We went with the name because on the if statement it reads clearly as "If this cannot generate plate" but i couldn't think of a better alternative name for it